cli/daemon/run/testdata/echo_client/client.ts

```diff
@@ -184,7 +184,7 @@ export namespace echo {
 
     export interface EmptyData {
         OmitEmpty: Data<string, string>
-        NullPtr: string
+        NullPtr: (string | null)
```

Comment:
This change in Typescript is more correct as we're now modelling that these things could be null technically if a `nil` is set within the Go data structure.

This is because we do not validate if `nil` is set or not automatically

Reply:
True, but our recursive nature of how our types signatures are generated makes it very hard to do this without a large scale refactor

Reply:
That's true, but taken as a whole this change seems to make a bunch of changes to generated code that are (1) sometimes more incorrect, (2) sometimes more correct but unlikely to match user intent, and (3) more difficult to understand. 

These changes will cause compiler errors all over frontend application code.

On balance this doesn't seem like a good trade off. What are our options?

Reply:
> (1) sometimes more incorrect,
I disagree with this,

> (2) sometimes more correct but unlikely to match user intent, 
I disagree with this too, I think our existing behaviour is already unlikely to match user intent

> (3) more difficult to understand.
Only because I'm trying to keep this a minimal change set, I already have ideas on a better refactor to combine all type generators